### PR TITLE
CB-13293 : if fetch has been called already, no need to fetch again

### PR DESCRIPTION
**Pull request by audreyso**
### What does this PR do?

Prevent fetch from being called, if it has already been called.

### Checklist
- [X] [Reported an issue](http://cordova.apache.org/contribute/issues.html) in the JIRA database
- [X] Commit message follows the format: "CB-3232: (android) Fix bug with resolving file paths", where CB-xxxx is the JIRA ID & "android" is the platform affected.
- [X] Added automated test coverage as appropriate for this change.

### CODE REVIEW

File: src/cordova/platform/addHelper.js [hunk 1]
```diff
@@ -292,7 +292,9 @@ function getVersionFromConfigFile (platform, cfg) {
 function downloadPlatform (projectRoot, platform, version, opts) {
     var target = version ? (platform + '@' + version) : platform;
     return Q().then(function () {
-        if (opts.fetch) {
+        var platformPath = path.join(projectRoot, 'platforms', platform);
+        var platformAlreadyAdded = fs.existsSync(platformPath);
+        if ((opts.fetch) && (!platformAlreadyAdded || platformAlreadyAdded === undefined || platformAlreadyAdded === null)) {
```
Comment (purplecabbage) [inline src/cordova/platform/addHelper.js]: If platformAlreadyAdded is undefined or null the first condition will be truthy so there is no point in adding the additional tests.

Comment (audreyso) [reply src/cordova/platform/addHelper.js]: Okay. Thank you! I will remove it.

File: src/cordova/platform/addHelper.js [hunk 2]
```diff
@@ -292,7 +292,9 @@ function getVersionFromConfigFile (platform, cfg) {
 function downloadPlatform (projectRoot, platform, version, opts) {
     var target = version ? (platform + '@' + version) : platform;
     return Q().then(function () {
-        if (opts.fetch) {
+        var platformPath = path.join(projectRoot, 'platforms', platform);
+        var platformAlreadyAdded = fs.existsSync(platformPath);
+        if ((opts.fetch) && (!platformAlreadyAdded)) {
```
Comment (stevengill) [inline src/cordova/platform/addHelper.js]: So what do we return if `platformAlreadyAdded` is true? I think we need to return something to make sure we get to line 333. 

In line 333, what is `libDir` if it does fetch?

Comment (audreyso) [reply src/cordova/platform/addHelper.js]: For instance, if I add the ios platform to my project, libDir = /Users/auso/cordova/tests/project_9/node_modules/cordova-ios

Comment (stevengill) [reply src/cordova/platform/addHelper.js]: in line 333, `libDir` is defined?

Comment (audreyso) [reply src/cordova/platform/addHelper.js]: Yes! These are my logs:
`Error: Platform ios already added.
libDir = /Users/auso/.cordova/lib/npm_cache/cordova-ios/4.5.4/package`

Comment (stevengill) [reply src/cordova/platform/addHelper.js]: is your manual test using `fetch:true`

Comment (stevengill) [reply src/cordova/platform/addHelper.js]: Lets wait and rebase this after we merge your commit removing no-fetch

Comment (audreyso) [reply src/cordova/platform/addHelper.js]: Just rebased!

**Comment by raphinesse** [comment]
This is made obsolete by https://github.com/apache/cordova-fetch/pull/24. Closing.

**Pull request closed without merging**